### ISSUE TITLE
Several questions about the video streaming.

### ISSUE THREAD

**Issue by godka** [SIGCOMM21-5G/artifact#8]
Hi all, 

congrats on your excellent paper. Recently I'm investigating the reason why RL-based ABR didn't work In 5G, as reported in Fig. 17. However, I find some results, which is detailed as follows:

i) The results on BOLA show unexpected buffer increases. Here's an example from 'log_BOLA_5g_trace_104_driving' in line 21:

```
1606444848.4	40000	1.690093	0.0	13274781	2247	20.0
1606444859.13	40000	35.104808	0.102	2222640	385	23.68
```

We can see the buffer occupancy seems increased by 34 seconds without any additional operation.
While considering that the previous chunk spent almost 2.2 seconds to download completely, we doubt that there may occur an interrupt stall event here.
Thus, we suspect that the video player has been pressed the pause button here.

The rest of the cases are listed as follows (i.e., filename, line):

```
log_BOLA_5g_trace_105_driving 23
log_BOLA_5g_trace_105_driving 30
log_BOLA_5g_trace_105_driving 57
log_BOLA_5g_trace_68_driving 21
log_BOLA_5g_trace_68_driving 56
log_BOLA_5g_trace_78_driving 5
log_BOLA_5g_trace_78_driving 13
log_BOLA_5g_trace_78_driving 34
log_BOLA_5g_trace_71_driving 1
log_BOLA_5g_trace_71_driving 22
log_BOLA_5g_trace_95_driving 10
log_BOLA_5g_trace_95_driving 28
log_BOLA_5g_trace_95_driving 73
log_BOLA_5g_trace_85_driving 19
log_BOLA_5g_trace_85_driving 22
log_BOLA_5g_trace_70_driving 8
log_BOLA_5g_trace_70_driving 57
log_BOLA_5g_trace_70_driving 62
log_fastMPC_5g_trace_70_driving 27
log_BOLA_5g_trace_94_driving 8
log_BOLA_5g_trace_94_driving 9
log_BOLA_5g_trace_94_driving 10
log_BOLA_5g_trace_94_driving 22
log_BOLA_5g_trace_94_driving 23
log_BOLA_5g_trace_94_driving 78
log_BOLA_5g_trace_1_driving 4
log_BOLA_5g_trace_1_driving 11
log_BOLA_5g_trace_1_driving 14
log_BOLA_5g_trace_1_driving 45
log_BOLA_5g_trace_1_driving 47
log_BOLA_5g_trace_104_driving 3
log_BOLA_5g_trace_104_driving 15
log_BOLA_5g_trace_104_driving 16
log_BOLA_5g_trace_104_driving 17
log_BOLA_5g_trace_104_driving 29
log_BOLA_5g_trace_79_driving 20
log_BOLA_5g_trace_79_driving 22
log_BOLA_5g_trace_79_driving 72
log_BOLA_5g_trace_69_driving 11
log_BOLA_5g_trace_69_driving 16
log_BOLA_5g_trace_69_driving 30
log_BOLA_5g_trace_69_driving 68
log_BOLA_5g_trace_24_driving 20
log_BOLA_5g_trace_24_driving 43
log_BOLA_5g_trace_24_driving 69
log_BOLA_5g_trace_72_driving 16
log_BOLA_5g_trace_72_driving 21
log_BOLA_5g_trace_72_driving 37
log_BOLA_5g_trace_96_driving 10
log_BOLA_5g_trace_96_driving 35
log_BOLA_5g_trace_96_driving 88
log_BOLA_5g_trace_3_driving 3
log_BOLA_5g_trace_3_driving 31
log_BOLA_5g_trace_3_driving 34
log_BOLA_5g_trace_37_driving 23
log_BOLA_5g_trace_37_driving 84
log_BOLA_5g_trace_73_driving 8
log_BOLA_5g_trace_73_driving 19
log_BOLA_5g_trace_73_driving 25
log_BOLA_5g_trace_2_driving 2
log_BOLA_5g_trace_2_driving 18
log_BOLA_5g_trace_2_driving 90
log_BOLA_5g_trace_97_driving 24
log_BOLA_5g_trace_7_driving 14
log_BOLA_5g_trace_7_driving 22
log_BOLA_5g_trace_7_driving 120
log_BOLA_5g_trace_59_driving 11
log_BOLA_5g_trace_59_driving 114
log_BOLA_5g_trace_59_driving 119
log_BOLA_5g_trace_76_driving 8
log_BOLA_5g_trace_76_driving 9
log_BOLA_5g_trace_66_driving 9
log_BOLA_5g_trace_22_driving 24
log_BOLA_5g_trace_22_driving 32
log_BOLA_5g_trace_22_driving 38
log_BOLA_5g_trace_22_driving 61
log_BOLA_5g_trace_23_driving 19
log_BOLA_5g_trace_23_driving 40
log_BOLA_5g_trace_103_driving 10
log_BOLA_5g_trace_103_driving 23
log_BOLA_5g_trace_103_driving 30
log_BOLA_5g_trace_93_driving 14
log_BOLA_5g_trace_93_driving 22
log_BOLA_5g_trace_6_driving 8
log_BOLA_5g_trace_6_driving 13
log_BOLA_5g_trace_6_driving 15
log_BOLA_5g_trace_6_driving 28
log_BOLA_5g_trace_6_driving 31
log_BOLA_5g_trace_58_driving 13
log_BOLA_5g_trace_58_driving 56
log_BOLA_5g_trace_58_driving 58
log_BOLA_5g_trace_58_driving 71
log_BOLA_5g_trace_15_driving 2
log_BOLA_5g_trace_15_driving 12
log_BOLA_5g_trace_15_driving 27
log_BOLA_5g_trace_15_driving 104
log_BOLA_5g_trace_67_driving 2
log_BOLA_5g_trace_67_driving 10
log_BOLA_5g_trace_67_driving 22
log_BOLA_5g_trace_67_driving 29
log_BOLA_5g_trace_77_driving 8
log_BOLA_5g_trace_77_driving 11
log_BOLA_5g_trace_77_driving 20
log_BOLA_5g_trace_88_driving 9
log_BOLA_5g_trace_88_driving 51
log_BOLA_5g_trace_98_driving 28
log_BOLA_5g_trace_98_driving 29
log_BOLA_5g_trace_98_driving 30
log_BOLA_5g_trace_98_driving 42
log_BOLA_5g_trace_98_driving 54
log_BOLA_5g_trace_43_driving 3
log_BOLA_5g_trace_43_driving 18
log_BOLA_5g_trace_43_driving 21
log_BOLA_5g_trace_43_driving 33
log_BOLA_5g_trace_43_driving 45
log_BOLA_5g_trace_81_driving 2
log_BOLA_5g_trace_81_driving 5
log_BOLA_5g_trace_81_driving 12
log_BOLA_5g_trace_81_driving 27
log_BOLA_5g_trace_81_driving 49
log_BOLA_5g_trace_38_driving 10
log_BOLA_5g_trace_38_driving 23
log_BOLA_5g_trace_38_driving 33
log_BOLA_5g_trace_118_driving 3
log_BOLA_5g_trace_118_driving 4
log_BOLA_5g_trace_118_driving 11
log_BOLA_5g_trace_118_driving 22
log_BOLA_5g_trace_118_driving 55
log_BOLA_5g_trace_39_driving 10
log_BOLA_5g_trace_39_driving 14
log_BOLA_5g_trace_39_driving 30
log_BOLA_5g_trace_39_driving 34
log_BOLA_5g_trace_80_driving 21
log_BOLA_5g_trace_80_driving 38
log_BOLA_5g_trace_80_driving 56
log_BOLA_5g_trace_119_driving 18
log_BOLA_5g_trace_119_driving 52
log_BOLA_5g_trace_109_driving 20
log_BOLA_5g_trace_109_driving 21
log_BOLA_5g_trace_109_driving 107
log_BOLA_5g_trace_89_driving 10
log_BOLA_5g_trace_89_driving 14
log_BOLA_5g_trace_89_driving 26
log_BOLA_5g_trace_89_driving 52
log_BOLA_5g_trace_42_driving 21
log_BOLA_5g_trace_42_driving 36
log_BOLA_5g_trace_110_driving 19
```

ii) The results of Pensieve may be incorrect since the RL server will not return the actual chunk-size after the video index is greater than 48:

https://github.com/SIGCOMM21-5G/artifact/blob/2bb3724df08666b8f57ca493a3f97707bb4bc485/Video-Streaming/ABR-5G/rl_server/rl_server_no_training.py#L61

To that end, Pensieve will feel confused if the future video chunks are all 0. Moreover, is it possible to provide the retrained Pensieve model for in-depth research? In turn, such codes are correct in the RobustMPC server:

https://github.com/SIGCOMM21-5G/artifact/blob/2bb3724df08666b8f57ca493a3f97707bb4bc485/Video-Streaming/ABR-5G/rl_server/robust_mpc_server.py#L74

Thanks to all the authors again.

Best,
Tianchi

**Comment by ry4nzhu** [comment]
Hi, thanks a lot for you constructive feedback.
For question ii), yes that is a good catch (we did a lot of test on different total chunks of the video and I forget to change it back). I've rerun the Pensieve experiment and will create a PR to update the results. The results looks pretty similar after getting `index > TOTAL_VIDEO_CHUNKS` fixed. I'm attaching the updated figure here. 

![image](https://user-images.githubusercontent.com/44684279/129972865-c63b3bb4-5305-4f22-9517-7fc2780cdd78.png)


For i), the BOLA implementation we used is provided by dash.js (an earlier version) and we adapted the framework from the [pensieve](https://github.com/hongzimao/pensieve) repo, it's possible that some chunk information logging is skipped by the codebase as it is quite complex. We have tested with video GUI enabled and we have not observe any abnormal stopping events like the video player has been pressed the pause button during our experiments.
